lib/tests/streamlit/type_util_test.py

```diff
@@ -712,6 +715,127 @@ def test_convert_df_with_missing_values(self):
         )
 
 
+class TestArrowTruncation(DeltaGeneratorTestCase):
+    """Test class for the automatic arrow truncation feature."""
+
+    @patch_config_options(
+        {"server.maxMessageSize": 3, "server.enableArrowTruncation": True}
+    )
+    def test_truncate_larger_table(self):
+        """Test that `_maybe_truncate_table` correctly truncates a table that is
+        larger than the max message size.
+        """
+        col_data = list(range(200000))
+        original_df = pd.DataFrame(
+            {
+                "col 1": col_data,
+                "col 2": col_data,
+                "col 3": col_data,
+            }
+        )
+
+        original_table = pa.Table.from_pandas(original_df)
+        truncated_table = type_util._maybe_truncate_table(
+            pa.Table.from_pandas(original_df)
+        )
+        from streamlit import config
+
+        print(
+            config.get_option("server.enableArrowTruncation"),
+            config.get_option("server.maxMessageSize"),
+            truncated_table.nbytes,
+            original_table.nbytes,
+            truncated_table.num_rows,
+            original_table.num_rows,
+            pa.__version__,
+            pa.Table.from_pandas(original_df).slice(0, 100).nbytes,
+            pa.Table.from_pandas(truncated_table.to_pandas()).nbytes,
+        )
```

Comment:
nit: leftover debug print

Reply:
ups, fixed. This would be awesome as a linter rule